oak_functions/loader/src/attestation.rs

```diff
@@ -226,3 +177,39 @@ where
         Ok(Response::new(Box::pin(response_stream)))
     }
 }
+
+/// Attest a single gRPC streaming request. Client messages are provided via `receiver`.
+async fn attest(
+    receiver: &mut Receiver,
+    tee_certificate: &[u8],
+) -> anyhow::Result<(AttestedInvokeResponse, AttestedSession)> {
+    let request = receiver
+        .receive()
+        .await
+        .context("Couldn't receive attestation request")?
+        .context("Stream stopped preemptively")?;
+
+    // Receive client's attestation request containing a public key.
+    let request_type = request.request_type.context("Couldn't read request type")?;
+    let client_identity = if let RequestType::ClientIdentity(request) = request_type {
+        Ok(request)
```

Comment:
I don't think this should be wrapped in an Ok just to be unwrapped immediately. I would just make this branch return `request`, and use `anyhow::bail` or an explicit return in the other branch. Up to you.